controllers/front/OrderConfirmationController.php

```diff
@@ -158,6 +165,10 @@ protected function checkFreeOrder()
             false,
             $cart->secure_key
         );
+
+        if ($order->currentOrder) {
+            $this->isFreeOrder = true;
+        }
```

Comment:
```suggestion
```

Reply:
If order has been properly created I want to mark this controller as one working with free order, I think this condition is all right